ICAFIX/scripts/RecleanClassify.sh

```diff
@@ -0,0 +1,284 @@
+tempfiles_add "$LanguageROItemp"_labels.dlabel.nii
+wb_command -cifti-label-import "$roitemp" "$HCPPIPEDIR"/global/config/Language.ROI.txt "$LanguageROItemp"_labels.dlabel.nii -discard-others
+wb_command -cifti-math 'x > 0' "$LanguageROItemp" -var x "$LanguageROItemp"_labels.dlabel.nii
+
+# shortcut in case the folder gets renamed
+this_script_dir=$(dirname "$0")
+HelpFuncPath="$this_script_dir/scripts"
+#all arguments are strings, so we can can use the same argument list for compiled and interpreted
+matlab_argarray=("$StudyFolder" "$Subject" "$fMRIListName" "$subjectExpectedTimepoints" "$HighPass" "$fMRIResolution" "$GrayOrdinateTemplateFile" "$CorticalParcellationFile" "$roisubtemp" "$WMLabelFile" "$CSFLabelFile" "$VisualROItemp" "$LanguageROItemp" "$SubRegionsFile" "$NonGreyParcelsFile")
+
+case "$MatlabMode" in
+    (0)
+        matlab_cmd=("$this_script_dir/Compiled_computeRecleanFeatures/run_computeRecleanFeatures.sh" "$MATLAB_COMPILER_RUNTIME" "${matlab_argarray[@]}")
+        log_Msg "running compiled matlab command: ${matlab_cmd[*]}"
+        "${matlab_cmd[@]}"
+        ;;
+    (1 | 2)
+        #reformat argument array so matlab sees them as strings
+        matlab_args=""
+        for thisarg in "${matlab_argarray[@]}"
+        do
+            if [[ "$matlab_args" != "" ]]
+            then
+                matlab_args+=", "
+            fi
+            matlab_args+="'$thisarg'"
+        done
+        
+        matlabcode="
+            addpath('$HCPPIPEDIR/global/matlab');
+            addpath('$HCPPIPEDIR/global/scripts');
+            addpath('$HCPCIFTIRWDIR');
+            addpath('$this_script_dir/scripts');
+            addpath('$this_script_dir');
+            computeRecleanFeatures($matlab_args);"
+
+        log_Msg "running matlab code: $matlabcode"
+        "${matlab_interpreter[@]}" <<<"$matlabcode"
+        echo
+        ;;
+esac
+
+# hardcoded
+FixProbThresh="10"
+
+# inference each subjects under python environment
+for fMRIName in ${fMRINameToUse} ; do
+    RecleanFeaturePath="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/fix_reclean_features.csv"
+    FixProbPath="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica/fix_prob.csv"
+    PredictionResult="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${fMRIName}_hp${HighPass}.ica"
+
+    # ReclassifyAsSignalTxt="${PredictionResult}/ReclassifyAsSignalRecleanVote${VoteThresh}.txt"
+    # ReclassifyAsNoiseTxt="${PredictionResult}/ReclassifyAsNoiseRecleanVote${VoteThresh}.txt"
+
+    # ReclassifyAsSignalTxt="${PredictionResult}/${ReclassifyAsSignalFile}"
+    # ReclassifyAsNoiseTxt="${PredictionResult}/${ReclassifyAsNoiseFile}"
+
+    ReclassifyAsSignalTxt="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${ReclassifyAsSignalFile}"
+    ReclassifyAsNoiseTxt="${StudyFolder}/${Subject}/MNINonLinear/Results/${fMRIName}/${ReclassifyAsNoiseFile}"
```

Comment:
Again, please don't do this, just pass the absolute path as the argument in the first place.

Reply:
These two files cannot be added as arguments for this script, because they depend on fMRI Name.

Reply:
Oh...I see...so, because of SR FIX support for multiple fMRIs in a single call, they can't be absolute.  I can live with that for this script.